### PR TITLE
bal_addresses 0.9.9: bal_tools 0.1.2 + use new RPC class for permissions probing

### PR DESCRIPTION
**Pull request by Tritium-VLK**
This is testing something we can easily move to bal_tools.

### CODE REVIEW

**Comment by Tritium-VLK** [comment]
Goal is to import this from bal_tools or bal_addresses and work through a lot of the code to replace various implementations.  Figure out details here, then will finish this PR using what we decide:

https://github.com/BalancerMaxis/bal_tools/pull/27/

File: generate_current_permissions.py [hunk 1]
```diff
@@ -82,7 +93,7 @@ def generate_chain_files(chain):
 
 
 def main():
-    for chain in w3_by_chain:
+    for chain in AddrBook.chain_ids_by_name.keys():
```
Comment (gosuto-inzasheru) [inline generate_current_permissions.py:96]: i know this is just a draft, but im not sure about this; addr book holds a lot of chains that we dont consider production (ie optimism or goerli)

Comment (Tritium-VLK) [reply generate_current_permissions.py:96]: I think the idea is to provide an RPC if one exists.  Maybe we could test it and throw an exception here and now if it does not.

**Comment by Tritium-VLK** [comment]
Test run: https://github.com/BalancerMaxis/bal_addresses/actions/runs/10179690713

**Review by gosuto-inzasheru — APPROVED** [review]
runs good in ci

will set up another pr for bumping tools to latest v0.1.4 and adding fraxtal and mode to prod chains

**Pull request merged**